diff --git a/lib/bodyScrollLock.esm.js b/lib/bodyScrollLock.esm.js
index 7a122dac144a72ef24f7fda3c59964e66a205051..8f0d704218ffa1eff4d2f3415da131c50210c4dd 100644
--- a/lib/bodyScrollLock.esm.js
+++ b/lib/bodyScrollLock.esm.js
@@ -64,22 +64,22 @@ var setOverflowHidden = function setOverflowHidden(options) {
     var scrollBarGap = window.innerWidth - document.documentElement.clientWidth;
 
     if (_reserveScrollBarGap && scrollBarGap > 0) {
-      var computedBodyPaddingRight = parseInt(window.getComputedStyle(document.body).getPropertyValue('padding-right'), 10);
-      previousBodyPaddingRight = document.body.style.paddingRight;
-      document.body.style.paddingRight = computedBodyPaddingRight + scrollBarGap + 'px';
+      const computedBodyPaddingRight = parseInt(window.getComputedStyle(window.top.document.body).getPropertyValue('padding-right'), 10);
+      previousBodyPaddingRight = window.top.document.body.style.paddingRight;
+      window.top.document.body.style.paddingRight = `${computedBodyPaddingRight + scrollBarGap}px`;
     }
   }
 
   // If previousBodyOverflowSetting is already set, don't set it again.
   if (previousBodyOverflowSetting === undefined) {
-    previousBodyOverflowSetting = document.body.style.overflow;
-    document.body.style.overflow = 'hidden';
+    previousBodyOverflowSetting = window.top.document.body.style.overflow;
+    window.top.document.body.style.overflow = 'hidden';
   }
 };
 
 var restoreOverflowSetting = function restoreOverflowSetting() {
   if (previousBodyPaddingRight !== undefined) {
-    document.body.style.paddingRight = previousBodyPaddingRight;
+    window.top.document.body.style.paddingRight = previousBodyPaddingRight;
 
     // Restore previousBodyPaddingRight to undefined so setOverflowHidden knows it
     // can be set again.
@@ -87,7 +87,7 @@ var restoreOverflowSetting = function restoreOverflowSetting() {
   }
 
   if (previousBodyOverflowSetting !== undefined) {
-    document.body.style.overflow = previousBodyOverflowSetting;
+    window.top.document.body.style.overflow = previousBodyOverflowSetting;
 
     // Restore previousBodyOverflowSetting to undefined
     // so setOverflowHidden knows it can be set again.
@@ -100,31 +100,16 @@ var setPositionFixed = function setPositionFixed() {
     // If previousBodyPosition is already set, don't set it again.
     if (previousBodyPosition === undefined) {
       previousBodyPosition = {
-        position: document.body.style.position,
-        top: document.body.style.top,
-        left: document.body.style.left
+        position: window.top.body.style.position,
+        top: window.top.body.style.top,
+        left: window.top.body.style.left
       };
 
       // Update the dom inside an animation frame 
-      var _window = window,
-          scrollY = _window.scrollY,
-          scrollX = _window.scrollX,
-          innerHeight = _window.innerHeight;
-
-      document.body.style.position = 'fixed';
-      document.body.style.top = -scrollY;
-      document.body.style.left = -scrollX;
-
-      setTimeout(function () {
-        return window.requestAnimationFrame(function () {
-          // Attempt to check if the bottom bar appeared due to the position change
-          var bottomBarHeight = innerHeight - window.innerHeight;
-          if (bottomBarHeight && scrollY >= innerHeight) {
-            // Move the content further up so that the bottom bar doesn't hide it
-            document.body.style.top = -(scrollY + bottomBarHeight);
-          }
-        });
-      }, 300);
+      const { scrollY, scrollX, innerHeight } = window;
+      window.top.document.body.style.position = 'fixed';
+      window.top.document.body.style.top = `${-scrollY}px`;
+      window.top.document.body.style.left = `${-scrollX}px`;
     }
   });
 };
@@ -132,13 +117,13 @@ var setPositionFixed = function setPositionFixed() {
 var restorePositionSetting = function restorePositionSetting() {
   if (previousBodyPosition !== undefined) {
     // Convert the position from "px" to Int
-    var y = -parseInt(document.body.style.top, 10);
-    var x = -parseInt(document.body.style.left, 10);
+    const y = -parseInt(window.top.document.body.style.top, 10);
+    const x = -parseInt(window.top.document.body.style.left, 10);
 
     // Restore styles
-    document.body.style.position = previousBodyPosition.position;
-    document.body.style.top = previousBodyPosition.top;
-    document.body.style.left = previousBodyPosition.left;
+    window.top.body.style.position = previousBodyPosition.position;
+    window.top.body.style.top = previousBodyPosition.top;
+    window.top.body.style.left = previousBodyPosition.left;
 
     // Restore scroll
     window.scrollTo(x, y);