diff --git a/index.js b/index.js
index 7d523be39ed5ad53c54b39fc106b137b32b9d602..69318951a0f3096c47143aede1c50c94fe575fbd 100644
--- a/index.js
+++ b/index.js
@@ -223,7 +223,7 @@ export default class Emittery {
 
 		// eslint-disable-next-line n/prefer-global/process
 		const {env} = globalThis.process || {env: {}};
-		return env.DEBUG === 'emittery' || env.DEBUG === '*' || isGlobalDebugEnabled;
+		return env?.DEBUG === 'emittery' || env?.DEBUG === '*' || isGlobalDebugEnabled;
 	}
 
 	static set isDebugEnabled(newValue) {